src/pages/Causes/Signup.vue

```diff
@@ -1,154 +1,111 @@
+	my {
+		subscriptions {
+			values {
+				id
+				subscrId
+			}
+		}
+		autoDeposit {
+			id
+			isSubscriber
+		}
+		lender {
+			id
+			loans {
+				totalCount
+			}
+		}
+	}
+}`;
 
 export default {
-	components: {
-		KvButton,
-		KvGrid,
-		KvPageContainer,
-		KvTextInput,
-	},
-	metaInfo: {
-		title: 'Get Started',
-	},
-	mixins: [validationMixin],
-	validations: {
-		email: {
-			required,
-			email,
+	props: {
+		cause: {
+			type: String,
+			default: 'climate',
+			validator: value => {
+				return ['climate', 'education', 'women'].indexOf(value) !== -1;
+			}
 		},
 	},
+	components: {
+		SignupForm,
+		SignupWaitlist,
+	},
 	data() {
 		return {
-			email: null,
-			step: 'join',
-			causesIconImgRequire,
+			// user flags
+			isMonthlyGoodSubscriber: false,
+			hasAutoDeposits: false,
+			hasLegacySubscription: false,
+			hasMadeLoan: false,
 		};
 	},
-	inject: ['apollo'],
-	methods: {
-		async submitForm() {
-			const isProd = window.location.hostname === 'www.kiva.org';
-			const iterableListIdString = isProd
-				? '1a075918-42c4-49f8-a3e9-e797dcf7c9b4'
-				: 'bacb00cb-ae81-4ab6-8981-b4fafb2026ce';
-
-			this.$v.$touch();
-			if (!this.$v.$invalid) {
-				// Track facebook event
-				if (typeof window !== 'undefined' && typeof fbq === 'function') {
-					window.fbq('track', 'Contact');
+	inject: ['apollo', 'cookieStore'],
+	apollo: {
+		preFetch(config, client, { route }) {
+			return client.query({
+				query: authenticationQuery,
+				fetchPolicy: 'network-only',
+			}).then(({ data }) => {
+				if (!data.my) {
+					throw new Error('api.authenticationRequired');
 				}
-
-				// eslint-disable-next-line max-len
-				const response = await fetch(`//links.iterable.com/lists/publicAddSubscriberForm?publicIdString=${iterableListIdString}`, {
-					method: 'POST',
-					body: new URLSearchParams({
-						email: this.email,
-					})
+				return client.query({ query: pageQuery });
+			}).catch(() => {
+				// Auth error will be caught here, redirect to login.
+				return Promise.reject({
+					path: `/ui-login?cause=${this.cause}`,
+					query: { force: true, doneUrl: route.fullPath }
 				});
-				if (response.status === 200) {
-					this.step = 'thanks';
-				} else {
-					this.$showTipMsg('An Error has occured. Please refresh the page and try again.', 'error');
-				}
-			}
+			});
+		}
+	},
+	created() {
+		try {
+			const pageQueryResult = this.apollo.readQuery({
+				query: pageQuery,
+			});
+			this.isMonthlyGoodSubscriber = pageQueryResult?.my?.autoDeposit?.isSubscriber ?? false;
+			this.hasAutoDeposits = pageQueryResult?.my?.autoDeposit ?? false;
+			const legacySubs = pageQueryResult?.my?.subscriptions?.values ?? [];
+			this.hasLegacySubscription = legacySubs.length > 0;
+			this.hasMadeLoan = pageQueryResult?.my?.lender?.loanCount > 0;
+		} catch (e) {
+			logReadQueryError(e, 'Causes Signup causesSignupEligibilityQuery');
+		}
+	},
+	computed: {
+		// Temporary query param feature flag
```

Comment:
This query param flag is just so we can test this and validate in lower environments